src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.cs

```diff
@@ -627,4 +627,58 @@ public static IEndpointRouteBuilder MapPost(this IEndpointRouteBuilder app, int
         var result = Assert.IsType<GeneratorRunResult>(Assert.Single(generatorRunResult.Results));
         Assert.Empty(GetStaticEndpoints(result, GeneratorSteps.EndpointModelStep));
     }
+
+    [Fact]
+    public async Task TestHandlingOfGenericWithNullableReferenceTypes()
+    {
+        var source = """
+using System;
+using System.Globalization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+
+namespace TestApp
+{
+    public static class TestMapActions
+    {
+        public static IEndpointRouteBuilder MapTestEndpoints(this IEndpointRouteBuilder app)
+        {
+            app.MapGet("/", (IGenericService<SomeInput, string?> service)
+                => "Maybe? " + service.Get(new SomeInput(Random.Shared.Next())));
+            return app;
+        }
+    }
+
+    public interface IInputConstraint<TOutput>;
+
+    public interface IGenericService<TInput, TOutput> where TInput : IInputConstraint<TOutput>
+    {
+        TOutput Get(TInput input);
+    }
+
+    public record SomeInput(int Value) : IInputConstraint<string?>;
+
+    public class ConcreteService : IGenericService<SomeInput, string?>
+    {
+        public string? Get(SomeInput input) => input.Value % 2 == 0 ? input.Value.ToString(CultureInfo.InvariantCulture) : null;
+    }
+}
+""";
+        var project = CreateProject();
+        project = project.AddDocument("TestMapActions.cs", SourceText.From(source, Encoding.UTF8)).Project;
+        var compilation = await project.GetCompilationAsync();
+
+        var generator = new RequestDelegateGenerator.RequestDelegateGenerator().AsSourceGenerator();
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(generators: new[]
+            {
+                generator
+            },
+            driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true),
+            parseOptions: ParseOptions);
+        driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out var updatedCompilation,
+            out var diagnostics);
+
+        var diagnosticsR = updatedCompilation.GetDiagnostics();
+        Assert.Empty(diagnosticsR.Where(d => d.Severity >= DiagnosticSeverity.Warning));
```

Comment:
I'm guessing this is a typo.
```suggestion
        var diagnostics = updatedCompilation.GetDiagnostics();
        Assert.Empty(diagnostics.Where(d => d.Severity >= DiagnosticSeverity.Warning));
```